<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Bayesian Enrichment Estimation in R (BEER)</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />
<link rel="apple-touch-icon" sizes="180x180" href="site_libs/images/beer_icons/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="site_libs/images/beer_icons/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="site_libs/images/beer_icons/favicon-16x16.png">
<link rel="manifest" href="site_libs/images/beer_icons/site.webmanifest">
<link rel="mask-icon" href="site_libs/images/beer_icons/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#b91d47">
<meta name="theme-color" content="#ffffff">

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">beer_manuscript</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="beer.html">BEER</a>
</li>
<li>
  <a href="simulation.html">Simulated Data</a>
</li>
<li>
  <a href="hiv_ec.html">HIV Elite Controllers</a>
</li>
<li>
  <a href="coronascan.html">CoronaScan Samples</a>
</li>
<li>
  <a href="hiv_discussion.html">HIV Discussion</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://www.biorxiv.org/content/10.1101/2022.01.19.476926v1">
    <span class="fa fa-file-alt"></span>
     
    Manuscript
  </a>
</li>
<li>
  <a href="https://github.com/athchen/beer_manuscript">
    <span class="fa fa-github"></span>
     
    Source Code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Bayesian Enrichment Estimation in R (BEER)</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>beer_manuscript/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210907code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210907)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210907code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210907)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomathchenbeermanuscripttreec46876e86dc9288a409eb1777629ff4613a1ae5ftargetblankc46876ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/athchen/beer_manuscript/tree/c46876e86dc9288a409eb1777629ff4613a1ae5f" target="_blank">c46876e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomathchenbeermanuscripttreec46876e86dc9288a409eb1777629ff4613a1ae5ftargetblankc46876ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/athchen/beer_manuscript/tree/c46876e86dc9288a409eb1777629ff4613a1ae5f" target="_blank">c46876e</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/partials/.DS_Store
    Ignored:    data_processed/.DS_Store
    Ignored:    data_processed/simulation_2beads_mle/.DS_Store
    Ignored:    data_processed/simulation_2beads_mom/.DS_Store
    Ignored:    data_processed/simulation_2beads_truth/.DS_Store
    Ignored:    data_processed/simulation_4beads_edgeR/.DS_Store
    Ignored:    data_processed/simulation_curves.rda
    Ignored:    data_raw/.DS_Store
    Ignored:    figures/.DS_Store

Untracked files:
    Untracked:  R/figure_postpred.R
    Untracked:  R/run_hiv_custom.R
    Untracked:  data_processed/hiv_samples.rds
    Untracked:  figures/postpred.png

Unstaged changes:
    Modified:   R/figure_attnconstant.R
    Modified:   R/figure_corstructure.R
    Modified:   R/figure_simulation_postprob.R
    Modified:   README.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/beer.Rmd</code>) and HTML (<code>docs/beer.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/athchen/beer_manuscript/blob/c46876e86dc9288a409eb1777629ff4613a1ae5f/analysis/beer.Rmd" target="_blank">c46876e</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-30
</td>
<td>
added Figure S13, moved figure
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/bbf7cba43c848a8c3c239a1ebce9c229941a1999/docs/beer.html" target="_blank">bbf7cba</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-23
</td>
<td>
rebuilt webpage
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/dc7a3fd47548194971ab58240961c6c65a7a362b/docs/beer.html" target="_blank">dc7a3fd</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/athchen/beer_manuscript/blob/01612f1d0ce7dd333592a3b08f2b3f75fd03d0f8/analysis/beer.Rmd" target="_blank">01612f1</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-23
</td>
<td>
updated model
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/3b06e484ffb5e477ff839bf86f222da54f3a6bd7/docs/beer.html" target="_blank">3b06e48</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/d8b023fb502b200a28b64e20a473e1d1520a2042/docs/beer.html" target="_blank">d8b023f</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/75f2a6d707f4b9253cdf10d8b7faa34ced412008/docs/beer.html" target="_blank">75f2a6d</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/athchen/beer_manuscript/blob/9d7b43ac86071aa909acd9f4d9807d3219c15b7e/analysis/beer.Rmd" target="_blank">9d7b43a</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-18
</td>
<td>
Updated repo to use most up-to-date figures and code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/9245fe24c20b4c6409f67ecf905ddf1ec87ce17a/docs/beer.html" target="_blank">9245fe2</a>
</td>
<td>
Athena Chen
</td>
<td>
2021-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/54af8302b1fc598ce31ce21096b4f880b7e7c0f4/docs/beer.html" target="_blank">54af830</a>
</td>
<td>
Athena Chen
</td>
<td>
2021-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/e30784bce4b9b6d5be2382063948b559b4f78f69/docs/beer.html" target="_blank">e30784b</a>
</td>
<td>
Athena Chen
</td>
<td>
2021-09-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/4984369ac084f50660e215c1733c6b29ef3e3b9a/docs/beer.html" target="_blank">4984369</a>
</td>
<td>
Athena Chen
</td>
<td>
2021-09-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/34893c7442935b4b7a4d9501bccec79fadcff0d2/docs/beer.html" target="_blank">34893c7</a>
</td>
<td>
Athena Chen
</td>
<td>
2021-09-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/athchen/beer_manuscript/blob/667d1afe73e64410a0d7924f55065ab99515c57e/analysis/beer.Rmd" target="_blank">667d1af</a>
</td>
<td>
Athena Chen
</td>
<td>
2021-09-14
</td>
<td>
Added simulation output
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/667d1afe73e64410a0d7924f55065ab99515c57e/docs/beer.html" target="_blank">667d1af</a>
</td>
<td>
Athena Chen
</td>
<td>
2021-09-14
</td>
<td>
Added simulation output
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>BEER is a Bayesian hierarchical model for identifying enriched antibody responses from phage-immunoprecipitation sequencing (PhIP-Seq) data. Here, we introduce the notation and describe the BEER model and implementation in the <code>R</code> package <a href="https://github.com/athchen/beer"><code>beer</code></a>.</p>
<div id="notation" class="section level2">
<h2>Notation</h2>
<p>Let <span class="math inline">\(i = 1, 2, \ldots, P\)</span> and <span class="math inline">\(j = 1, 2, \ldots, 96\)</span> index the peptides and samples, respectively. Without loss of generality, assume samples <span class="math inline">\(\{1, 2, \ldots, N\}\)</span> are mock IP (beads-only) samples. We observe,</p>
<p><span class="math display">\[\begin{align*}
Y_{ij} &amp;= \text{ observed # reads mapped to peptide } i \text{ in sample } j\\
n_j &amp;= \sum_{i=1}^P Y_{ij} \text{ total # of reads in sample }j
\end{align*}\]</span></p>
<p>Define,</p>
<p><span class="math display">\[\begin{align*}
\theta_{ij} &amp;= \text{ probability that peptide } i \text{ in sample } j \text{ pulls a read }\\
Z_{ij} &amp;= \unicode{x1D7D9}(\text{peptide } i \text{ in sample } j \text { is enriched})\\
\pi_{j} &amp;= \text{ proportion of enriched peptides in sample } j\\
c_j &amp;= \text{ attenuation constant for sample } j\\
\phi_{ij} &amp;= \text{ true fold change of peptide } i \text{ in sample } j
\end{align*}\]</span></p>
<p>Prior parameters for <span class="math inline">\(\theta_{ij}, \pi_j, c_j\)</span>, and <span class="math inline">\(\phi_{ij}\)</span> are denoted by,</p>
<p><span class="math display">\[\begin{align*}
a_{ij}, b_{ij} &amp;= \text{ shape parameters for the prior distribution of } \theta_{ij} \text{ for peptide } i \text{ in sample } j\\
a_{\pi}, b_{\pi} &amp;= \text{ shape parameters for the prior distribution of } \pi_{j}\\
a_c, b_c &amp;= \text{ shape parameters for the prior distribution of } c_{j}\\
a_\phi, b_\phi &amp;= \text{ shape parameters for the prior distribution of } \phi_{ij}|Z_{ij} = 1
\end{align*}\]</span></p>
<p>Parameters specific to beads-only samples are denoted with the subscript <span class="math inline">\(i0\)</span> (e.g. <span class="math inline">\(a_{i0}, b_{i0}, \theta_{i0}\)</span>, etc.). Additionally, let <span class="math inline">\(\phi_{min}\)</span> denote the minimum fold-change for an enriched peptide.</p>
<p>For convenience, we also define two function <span class="math inline">\(f_a, f_b\)</span> for deriving Beta parameters <span class="math inline">\(a, b\)</span> given mean <span class="math inline">\(\mu\)</span> and variance <span class="math inline">\(\sigma^2\)</span>:</p>
<p><span class="math display">\[\begin{align*}
f_a(\mu, \sigma^2) &amp;= \frac{\mu^2(1-\mu)}{\sigma^2} - \mu\\
f_b(\mu, \sigma^2) &amp;= f_a(\mu, \sigma^2) \left(\frac{1}{\mu} - 1\right)
\end{align*}\]</span></p>
</div>
<div id="model" class="section level2">
<h2>Model</h2>
<p>Let <span class="math inline">\(\mu_{i0}\)</span> and <span class="math inline">\(\sigma^2_{i0}\)</span> denote the mean and variance for peptide <span class="math inline">\(i\)</span> in a beads-only sample where,</p>
<p><span class="math display">\[\begin{align*}
\mu_{ij} &amp;= \frac{a_{i0}}{a_{i0} + b_{i0}}\\
\sigma^2_{i0} &amp;= \frac{a_{i0}b_{i0}}{(a_{i0} + b_{i0})^2 (a_{i0} + b_{i0} + 1)}.
\end{align*}\]</span></p>
<p>To infer reactivity, we compare one sample to all beads-only samples on the same plate. Our hierarchical model given a sample <span class="math inline">\(j \in \{N+1, \ldots, 96\}\)</span> is described as follows.</p>
<p><span class="math display">\[\begin{align*}
    Y_{ij}|\theta_{ij} &amp;\sim \text{Binomial}(n_j, \theta_{ij}) \\
    \theta_{ij}|a_{i0}, b_{i0}, c_j, \phi_{ij} &amp;\sim \text{Beta}(f_a(c_j \phi_{ij} \mu_{i0}, \sigma^2_{i0}), f_b(c_j \phi_{ij} \mu_{i0}, \sigma^2_{i0})) \\
%    c_j|B_j &amp; \sim B_j\cdot 1 + (1 - B_j) \cdot \text{Beta}(a_c, b_c) \\
    c_j &amp; \sim \text{Beta}(a_c, b_c) \\
    \phi_{ij}|Z_{ij} &amp; \sim (1 - Z_{ij}) \cdot 1 + Z_{ij}(\phi_{min} + \text{Gamma}(a_\phi, b_\phi)) \\
    Z_{ij}|\pi_j &amp; \sim \text{Bernoulli}(\pi_j)\\
    \pi_j &amp;\sim \text{Beta}(a_\pi, b_\pi) 
\end{align*}\]</span></p>
</div>
<div id="prior-parameters" class="section level2">
<h2>Prior parameters</h2>
<div id="figure-s17" class="section level3">
<h3>Figure S17</h3>
<p><strong>Left:</strong> the prior distribution for the proportion of reactive peptides in sample <span class="math inline">\(j\)</span>, <span class="math inline">\(\pi_j\)</span>, modeled as a Beta distribution Beta(<span class="math inline">\(a_\pi\)</span> = 2, <span class="math inline">\(b_\pi\)</span> = 300), reflecting peptide enrichment seen in previous studies. <strong>Middle:</strong> a Gamma(<span class="math inline">\(a_\phi\)</span> = 1.25, <span class="math inline">\(b_\phi\)</span> = 0.1) distribution, used in the prior distribution for the fold change <span class="math inline">\(\phi_{ij}\)</span> for peptide <span class="math inline">\(i\)</span> in sample <span class="math inline">\(j\)</span>, if reactive. <strong>Right:</strong> the prior distribution for the scaling constant in sample <span class="math inline">\(j\)</span>, <span class="math inline">\(c_j\)</span>, modeled as a Beta distribution Beta(<span class="math inline">\(a_c = 80\)</span>, <span class="math inline">\(b_c = 20\)</span>).</p>
<pre class="r"><code>par(mfrow = c(1,3),las=1,yaxs=&quot;i&quot;,mar=c(5,1,3,1),cex.axis=1.3)
curve(dbeta(x,2,300),from=0,to=0.05,n=501,
      xlab=&quot;&quot;,ylab=&quot;&quot;,yaxt=&quot;n&quot;,ylim=c(0,115),lwd=3,col=&quot;red&quot;)
curve(dgamma(x,1.25,0.1),from=0,to=80,n=501,
      xlab=&quot;&quot;,ylab=&quot;&quot;,yaxt=&quot;n&quot;,ylim=c(0,0.065),lwd=3,col=&quot;blue&quot;)
curve(dbeta(x,80,20),from=0.5,to=1,n=501,
      xlab=&quot;&quot;,ylab=&quot;&quot;,yaxt=&quot;n&quot;,ylim=c(0,10.5),lwd=3,col=&quot;green3&quot;)</code></pre>
<p><img src="figure/beer.Rmd/priors-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-priors-1">
Past versions of priors-1.png
</button>
</p>
<div id="fig-priors-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/athchen/beer_manuscript/blob/75f2a6d707f4b9253cdf10d8b7faa34ced412008/docs/figure/beer.Rmd/priors-1.png" target="_blank">75f2a6d</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="implementation" class="section level2">
<h2>Implementation</h2>
<p>To reduce computational time, BEER runs each sample individually in comparison to all beads-only samples and removes clearly enriched peptides apriori. The implementation can be broken down into the following steps:</p>
<ol style="list-style-type: decimal">
<li><strong>Define prior parameters.</strong> Though most prior parameters are supplemented by the user (or use the defaults), prior parameters for non-enriched peptides are first approximated using all beads-only samples.</li>
<li><strong>Identify super enriched peptides.</strong> Based on the prior parameters, super enriched peptides are first excluded as these peptides should always have posterior probabilities of enrichment of 1.</li>
<li><strong>Re-estimate beads-only prior parameters.</strong> Prior parameters are then reestimated from the beads-only samples for the remaining peptides.</li>
<li><strong>Initialize and run the MCMCs.</strong> To reduce convergence time, MLE estimates are used to initialize the MCMC sampler, and samples are drawn from the posterior distributions of the unknown parameters.</li>
<li><strong>Summarize and store results.</strong> Posterior samples are summarized using the means of the posterior distribution and are stored in the PhIPData object.</li>
</ol>
<p>For more information, please see the <em>Supplemental Methods</em> section of the manuscript.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8       compiler_4.1.2   pillar_1.6.4     later_1.3.0     
 [5] git2r_0.29.0     jquerylib_0.1.4  highr_0.9        tools_4.1.2     
 [9] getPass_0.2-2    digest_0.6.29    evaluate_0.14    tibble_3.1.6    
[13] lifecycle_1.0.1  pkgconfig_2.0.3  rlang_0.4.12     rstudioapi_0.13 
[17] yaml_2.2.1       xfun_0.29        fastmap_1.1.0    httr_1.4.2      
[21] stringr_1.4.0    knitr_1.37       fs_1.5.2         vctrs_0.3.8     
[25] rprojroot_2.0.2  glue_1.6.0       R6_2.5.1         processx_3.5.2  
[29] fansi_1.0.2      rmarkdown_2.11   callr_3.7.0      magrittr_2.0.1  
[33] whisker_0.4      ps_1.6.0         promises_1.2.0.1 htmltools_0.5.2 
[37] ellipsis_0.3.2   httpuv_1.6.5     utf8_1.2.2       stringi_1.7.6   
[41] crayon_1.4.2    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
