<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>CoronaScan Samples</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />
<link rel="apple-touch-icon" sizes="180x180" href="site_libs/images/beer_icons/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="site_libs/images/beer_icons/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="site_libs/images/beer_icons/favicon-16x16.png">
<link rel="manifest" href="site_libs/images/beer_icons/site.webmanifest">
<link rel="mask-icon" href="site_libs/images/beer_icons/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#b91d47">
<meta name="theme-color" content="#ffffff">

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">beer_manuscript</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="beer.html">BEER</a>
</li>
<li>
  <a href="simulation.html">Simulated Data</a>
</li>
<li>
  <a href="hiv_ec.html">HIV Elite Controllers</a>
</li>
<li>
  <a href="coronascan.html">CoronaScan Samples</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="templink">
    <span class="fa fa-file-alt"></span>
     
    Manuscript
  </a>
</li>
<li>
  <a href="https://github.com/athchen/beer_manuscript">
    <span class="fa fa-github"></span>
     
    Source Code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">CoronaScan Samples</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-09-15
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>beer_manuscript/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210907code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210907)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210907code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210907)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomathchenbeermanuscripttree917bca796924d27da825226d043405ffc8c08c53targetblank917bca7a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/athchen/beer_manuscript/tree/917bca796924d27da825226d043405ffc8c08c53" target="_blank">917bca7</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomathchenbeermanuscripttree917bca796924d27da825226d043405ffc8c08c53targetblank917bca7a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/athchen/beer_manuscript/tree/917bca796924d27da825226d043405ffc8c08c53" target="_blank">917bca7</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/partials/.DS_Store
    Ignored:    data_processed/.DS_Store
    Ignored:    data_processed/simulation_2beads_mle/.DS_Store
    Ignored:    data_processed/simulation_2beads_mom/.DS_Store
    Ignored:    data_processed/simulation_2beads_truth/.DS_Store
    Ignored:    data_processed/simulation_4beads_edgeR/.DS_Store
    Ignored:    data_processed/simulation_curves.rda
    Ignored:    figures/.DS_Store

Unstaged changes:
    Modified:   R/figure_coronascan.R
    Modified:   README.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/coronascan.Rmd</code>) and HTML (<code>docs/coronascan.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/athchen/beer_manuscript/blob/917bca796924d27da825226d043405ffc8c08c53/analysis/coronascan.Rmd" target="_blank">917bca7</a>
</td>
<td>
Athena Chen
</td>
<td>
2021-09-15
</td>
<td>
wflow_publish(list.files(“analysis”, pattern = “Rmd”, full.names = TRUE))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/54af8302b1fc598ce31ce21096b4f880b7e7c0f4/docs/coronascan.html" target="_blank">54af830</a>
</td>
<td>
Athena Chen
</td>
<td>
2021-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/e30784bce4b9b6d5be2382063948b559b4f78f69/docs/coronascan.html" target="_blank">e30784b</a>
</td>
<td>
Athena Chen
</td>
<td>
2021-09-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/4984369ac084f50660e215c1733c6b29ef3e3b9a/docs/coronascan.html" target="_blank">4984369</a>
</td>
<td>
Athena Chen
</td>
<td>
2021-09-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/34893c7442935b4b7a4d9501bccec79fadcff0d2/docs/coronascan.html" target="_blank">34893c7</a>
</td>
<td>
Athena Chen
</td>
<td>
2021-09-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/athchen/beer_manuscript/blob/667d1afe73e64410a0d7924f55065ab99515c57e/analysis/coronascan.Rmd" target="_blank">667d1af</a>
</td>
<td>
Athena Chen
</td>
<td>
2021-09-14
</td>
<td>
Added simulation output
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/667d1afe73e64410a0d7924f55065ab99515c57e/docs/coronascan.html" target="_blank">667d1af</a>
</td>
<td>
Athena Chen
</td>
<td>
2021-09-14
</td>
<td>
Added simulation output
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>The CoronaScan PhIP-seq antigen library consists of 3466 unique peptides tiling proteins belonging to various coronaviruses, including SARS-CoV2, viruses associated with the common cold, and bat coronaviruses. Each peptide is present in duplicate, resulting in a total of 6932 peptides in the library.</p>
<p>This data set consists of eight beads-only samples and ten serum samples all run on the same plate. Of the ten serum samples, six samples are from individuals who participated in a vaccine research study prior to the discovery of SARS-CoV2. The remaining four samples are serum samples from various time points from one individual who was infected with SARS-CoV2. These samples were also tested for SARS-CoV2 antibodies.</p>
<pre class="r"><code>#&#39; Code to load required packages to reproduce the results and figures in 
#&#39; the manuscript. 
required_packages &lt;- c(&#39;plyr&#39;, &#39;tidyverse&#39;, &#39;here&#39;, &#39;ggpubr&#39;, &#39;gridExtra&#39;, 
                       &#39;latex2exp&#39;, &#39;kableExtra&#39;, &#39;RColorBrewer&#39;, &#39;BiocManager&#39;)
for (pkg in required_packages) {
    if (!(pkg %in% rownames(installed.packages()))) {
        install.packages(pkg)
    }
    library(pkg, character.only = TRUE)
}

bioc_packages &lt;- c(&quot;beer&quot;)
for(pkg in bioc_packages){
    if(!(pkg %in% rownames(installed.packages()))) {
        BiocManager::install(pkg)
    }
    library(pkg, character.only = TRUE)
}

rm(list = c(&quot;required_packages&quot;, &quot;bioc_packages&quot;, &quot;pkg&quot;))

#&#39; Define global variables for plotting
hot_cold_cols &lt;- c(&quot;navy&quot;, &quot;blue&quot;, &quot;deepskyblue&quot;, &quot;cyan&quot;, &quot;lightcyan&quot;,
                   &quot;yellow&quot;, &quot;orange&quot;, &quot;red&quot;)</code></pre>
<pre class="r"><code>#&#39; helper_functions.R 
#&#39; 
#&#39; functions used to help process and analyze output from BEER simulations

#&#39; integrate_vector()
#&#39; 
#&#39; Function to perform trapezoidal approximation given two vectors. 
#&#39; 
#&#39; @param x numeric vector of x values
#&#39; @param y numeric vector with the same length as x
#&#39; 
#&#39; @return numeric value of the trapezoidal approximation 
integrate_vector &lt;- function(x, y){
    
    if (length(y) != length(x)) {
        stop(&quot;The length of the vectors must be equal.&quot;)
    }
    
    n_points &lt;- length(y)
    sorted_ind &lt;- sort(x, index.return = TRUE)$ix
    sorted_x &lt;- x[sorted_ind]
    sorted_y &lt;- y[sorted_ind]
    
    # trapezoidal rule
    sum(0.5 * (sorted_y[-1] + sorted_y[-n_points]) * 
            (sorted_x[-1] - sorted_x[-n_points]))
}

#&#39; get_roc()
#&#39; 
#&#39; Function to calculate the ppv, sens, and spec for various
#&#39; cutoffs between 0 and 1. 
#&#39; 
#&#39; @param data data frame  with columns `prop_enriched` for the threshold and 
#&#39; `Z` for the true enrichment status, and optional column `extra_info`
#&#39; @param min_cutoff minimum cutoff, default to 0
#&#39; @param max_cutoff maximum cutoff, default ot 1 - 1e-6
#&#39; @param extra_info boolean indicating whether extra information should be used
#&#39; in the classification at each cutoff. If `TRUE`, then the column `extra_info`
#&#39; must be present in the data frame. 
#&#39; 
#&#39; @return data fram with columns `cutoff`, `ppv`, `sens`, and `spec`. 
get_roc &lt;- function(data, min_cutoff = 0, max_cutoff = 1 - 1e-6, 
                    extra_info = FALSE){
    
    cutoffs &lt;- seq(min_cutoff, max_cutoff, length.out = 1000)
    
    # Calculate ppv, sens, spec for each cutoff
    ppv &lt;- sapply(cutoffs, function(x) {
        if(extra_info){
            predict &lt;- (data$prop_enriched &gt;= x &amp; data$extra_info)
        } else {
            predict &lt;- (data$prop_enriched &gt;= x)
        }
        sum((data$Z == 1) &amp; predict)/sum(predict)
    })
    
    spec &lt;- sapply(cutoffs, function(x){
        if(extra_info){
            predict &lt;- (data$prop_enriched &gt;= x &amp; data$extra_info)
        } else {
            predict &lt;- (data$prop_enriched &gt;= x)
        }
        sum((data$Z == 0) &amp; !predict)/sum(data$Z == 0)
    })
    
    sens &lt;- sapply(cutoffs, function(x){
        if(extra_info){
            predict &lt;- (data$prop_enriched &gt;= x &amp; data$extra_info)
        } else {
            predict &lt;- (data$prop_enriched &gt;= x)
        }
        sum((data$Z == 1) &amp; predict)/sum(data$Z == 1)
    })
    
    # Get AUC for ROC
    npoints &lt;- length(sens)
    area_roc &lt;- integrate_vector(1-spec, sens)
    
    return(cutoffs = data.frame(cutoff = cutoffs,
                                ppv = ppv,
                                sens = sens,
                                spec = spec, 
                                area_roc = area_roc))
}

#&#39; get_legend()
#&#39; 
#&#39; Function given [here](https://stackoverflow.com/questions/12539348/ggplot-separate-legend-and-plot)
#&#39; to extract the legend for plotting purposes. 
#&#39; 
#&#39; @param plot ggplot
#&#39; @return ggplot legend
get_legend &lt;- function(myggplot){
    tmp &lt;- ggplot_gtable(ggplot_build(myggplot))
    leg &lt;- which(sapply(tmp$grobs, function(x) x$name) == &quot;guide-box&quot;)
    legend &lt;- tmp$grobs[[leg]]
    return(legend)
}

#&#39; mysqrt_trans()
#&#39; 
#&#39; Function given [here](https://stackoverflow.com/questions/47944992/ggplot2-removes-zero-when-using-scale-x-sqrt)
#&#39; to add zero to the plot after sqrt transforming the x-axis
#&#39; 
#&#39; @param plot ggplot
#&#39; @return ggplot legend
mysqrt_trans &lt;- function() {
    scales::trans_new(&quot;mysqrt&quot;, 
                      transform = base::sqrt,
                      inverse = function(x) ifelse(x&lt;0, 0, x^2),
                      domain = c(0, Inf))
}

#&#39; penriched_fit()
#&#39; 
#&#39; Function that returns a data frame with the point estimate, and 95\% 
#&#39; confidence intervals. 
#&#39; 
#&#39; @param model logistic regression model
#&#39; @param covariates data frame of covariates
#&#39; @return data frame with columns for the covariate, point estimate, lower CI
#&#39; and upper CI.
penriched_fit &lt;- function(model, covariates){
    
    # Predict based on the model
    prediction &lt;- predict(model, covariates, type = &quot;link&quot;, se.fit = TRUE)
    pred_lower &lt;- prediction$fit - 1.96*prediction$se.fit
    pred_upper &lt;- prediction$fit + 1.96*prediction$se.fit
    
    # Transform logit to probabilities
    ppred &lt;- 1/(1 + exp(-prediction$fit))
    plower &lt;- 1/(1 + exp(-pred_lower))
    pupper &lt;- 1/(1 + exp(-pred_upper))
    
    # Return covariates with prediction + 95 CI added
    bind_cols(covariates, 
              data.frame(predict_p = ppred, 
                         lower_ci = plower, 
                         upper_ci = pupper))
}</code></pre>
<pre class="r"><code>cs &lt;- readRDS(here(&quot;data_raw&quot;, &quot;coronascan.rds&quot;))

# Look at metadata
sampleInfo(cs)</code></pre>
<pre><code>DataFrame with 18 rows and 4 columns
              group         n days_since_infection     ab_test
        &lt;character&gt; &lt;numeric&gt;            &lt;numeric&gt; &lt;character&gt;
BEADS 1       beads    265705                   NA          NA
BEADS 2       beads    283164                   NA          NA
BEADS 3       beads    257012                   NA          NA
BEADS 4       beads    270956                   NA          NA
BEADS 5       beads    289627                   NA          NA
...             ...       ...                  ...         ...
CS 6            vrc    267894                   NA          NA
CS 7      SARS-CoV2    251115                   11         neg
CS 8      SARS-CoV2    246656                   10         neg
CS 9      SARS-CoV2    128815                   13         pos
CS 10     SARS-CoV2    255767                   12         neg</code></pre>
</div>
<div id="running-beer-and-edger" class="section level2">
<h2>Running BEER and edgeR</h2>
<p>To estimate the false-positive rate, we run each beads-only sample against the remaining seven beads-only samples.</p>
<pre class="r"><code>## Run edgeR with beadsRR
cs_out &lt;- edgeR(cs, assay.names = c(&quot;edgeR_logfc&quot;, &quot;edgeR_prob&quot;), 
                parallel = &quot;multisession&quot;, beadsRR = TRUE)

## Run beer with beadsRR
beer_assays &lt;- c(phi = NULL, phi_Z = &quot;beer_logfc&quot;, Z = &quot;beer_prob&quot;, 
                 c = &quot;sampleInfo&quot;, pi = &quot;sampleInfo&quot;)
cs_out &lt;- brew(cs_out, assay.names = beer_assays, beadsRR = TRUE, 
                parallel = &quot;multisession&quot;)</code></pre>
<p>For plotting convenience, we convert the PhIPData object to a tidy dataframe.</p>
<pre class="r"><code># Convert to tidy format and add hits for each approach
# Order samples by BEADS, VRC, then days since infection
cs_tidy &lt;- as(cs_out, &quot;DataFrame&quot;) %&gt;%
    as_tibble() %&gt;%
    group_by(sample) %&gt;%
    mutate(is_se = ifelse(sample != &quot;beads&quot; &amp; is.na(beer_prob), TRUE, FALSE), 
           sample = factor(sample, levels = colnames(cs)[c(1:14, 16, 15, 18, 17)]),
           beer_hits = ifelse(beer_prob &gt; 0.5 | is_se, 1, 0), 
           edgeR_bh = p.adjust(10^(-edgeR_prob), method = &quot;BH&quot;), 
           edgeR_hits = ifelse(edgeR_bh &lt; 0.05, 1, 0)) %&gt;%
    ungroup()</code></pre>
</div>
<div id="figures" class="section level2">
<h2>Figures</h2>
<div id="figure-s13" class="section level3">
<h3>Figure S13</h3>
<p>MA plots for the proportion of enriched peptides by protein for six pre-pandemic samples and four samples from one individual infected with SARS-CoV2. Samples from this individual were collected at various days since symptom onset (labels D10-D13) and were additionally tested for SARSCoV2 antibodies using (Ben) Ab test. Antibody test results (positive or negative) are indicated by Ab+ or Ab-, respectively. Points represent individual proteins; point colors indicate virus types; and point diameters indicate the number of peptides tiling the respective proteins. In the CoronaScan library, peptides are present in duplicate, so the number of peptides is double the number of unique peptides.</p>
<pre class="r"><code># Color palette
cs_species &lt;- unique(cs_tidy$organism)
grey_palette &lt;- palette(gray(seq(0.1, 0.8, len = (length(cs_species) - 1))))
# Make `grey_palette` is of the correct length. Not sure why this has to be run twice
grey_palette &lt;- if(length(grey_palette) &lt; (length(cs_species) - 1)){
    palette(gray(seq(0.1, 0.8, length.out = (length(cs_species) - 1))))
} else grey_palette
num_sarscov2 &lt;- grep(&quot;SARS-CoV2&quot;, cs_species)
cs_order &lt;- c(cs_species[num_sarscov2], cs_species[-num_sarscov2])

# Facet labels
facet_labels &lt;- case_when(
    sampleInfo(cs)$group == &quot;beads&quot; ~ colnames(cs), 
    sampleInfo(cs)$group == &quot;vrc&quot; ~ 
        paste0(&quot;VRC &quot;, gsub(&quot;CS &quot;, &quot;&quot;, colnames(cs))), 
    TRUE ~ paste0(&quot;SARS-CoV2, D&quot;, sampleInfo(cs)$days_since_infection, 
                  &quot; Ab&quot;, ifelse(sampleInfo(cs)$ab_test == &quot;pos&quot;, &quot;+&quot;, &quot;-&quot;)))
names(facet_labels) &lt;- colnames(cs)

cs_tidy %&gt;%
    filter(group != &quot;beads&quot;) %&gt;%
    select(sample, peptide, organism, protein_name, beer_hits, edgeR_hits) %&gt;%
    group_by(sample, organism, protein_name) %&gt;%
    summarize(prot_prop_Bayes = mean(beer_hits),
              prot_prop_edgeR = mean(edgeR_hits),
              num_peptides = n(), .groups = &quot;drop&quot;) %&gt;%
    mutate(organism = factor(organism, levels = cs_order)) %&gt;%
    arrange(desc(organism), desc(num_peptides)) %&gt;%
    ggplot(aes(x = 0.5*(prot_prop_Bayes + prot_prop_edgeR),
               y = prot_prop_Bayes - prot_prop_edgeR,
               color = organism,
               size = num_peptides/2)) +
    facet_wrap(sample ~., ncol = 5, 
               labeller = labeller(sample = facet_labels)) +
    geom_hline(aes(yintercept = 0), size = 0.5, color = &quot;grey50&quot;) +
    geom_vline(aes(xintercept = 0), size = 0.5, color = &quot;grey50&quot;) +
    geom_point(alpha = 0.8) +
    labs(x = TeX(&quot;$\\frac{1}{2}$ (BEER + edgeR)&quot;),
         y = &quot;BEER - edgeR&quot;,
         color = &quot;Organism&quot;,
         size = &quot;# peptides&quot;) +
    scale_x_continuous(trans = &quot;mysqrt&quot;, 
                       limits = c(0, 1), 
                       breaks = seq(0, 1, by = 0.2)) +
    scale_color_manual(values = c(grey_palette[1:(num_sarscov2 - 1)], 
                                  &quot;firebrick2&quot;, 
                                  grey_palette[-(1:(num_sarscov2 - 1))]), 
                       breaks = cs_species) +
    theme_bw() +
    theme(aspect.ratio = 1, 
          legend.title = element_text(size = 10), 
          legend.text = element_text(size = 8), 
          legend.key.size = unit(0.75, &quot;lines&quot;), 
          legend.position = &quot;bottom&quot;) +
    guides(color = guide_legend(ncol = 1, order = 1), 
           size = guide_legend(ncol = 1, order = 2))</code></pre>
<p><img src="figure/coronascan.Rmd/figure_coronascan_protein-1.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-s14" class="section level3">
<h3>Figure S14</h3>
<p>Concordance of paired peptides in CS sample 9. For each unique peptide sequence, the proportion of reads pulled for peptide 1 with the same sequence is plotted against the proportion of reads pulled for peptide 2 of the same sequence (top left). For BEER with edgeR estimates peptides are ranked by decreasing posterior probability of enrichment, with ties broken by the estimated fold-change. For edgeR, peptides are ranked by increasing p-values, with ties again broken by estimated fold-changes. The concordance between the rankings for the top <span class="math inline">\(k\)</span> ranks (x-axis) between all peptide 1’s and all peptide 2’s for each method is shown in the top right panel.</p>
<pre class="r"><code># Define sample to look at
cs_sample &lt;- &quot;CS 1&quot;

# Plot for proportion of reads
prop_reads &lt;- cs_tidy %&gt;% 
    filter(sample == cs_sample) %&gt;%
    mutate(prop_reads = counts/n) %&gt;%
    select(pair_id, pair_num, prop_reads) %&gt;%
    pivot_wider(names_from = &quot;pair_num&quot;, 
                values_from = &quot;prop_reads&quot;) %&gt;%
    dplyr::rename(pair_1 = `1`, pair_2 = `2`) %&gt;%
    ggplot(aes(x = log10(pair_1), 
               y = log10(pair_2))) + 
    geom_point(size = 1) +
    geom_abline(aes(intercept = 0, slope = 1)) + 
    labs(title = &quot;Proportion of reads pulled&quot;, 
         x = &quot;peptide pair #1, log10(proportion)&quot;, 
         y = &quot;peptide pair #2, log10(proportion)&quot;) +
    scale_x_continuous(breaks = seq(-5, -2, by = 1),
                       minor_breaks = seq(-5, -2, by = 0.5), 
                       labels = TeX(paste0(&quot;$10^{&quot;, -5:-2, &quot;}$&quot;))) +
    scale_y_continuous(breaks = seq(-5, -2, by = 1),
                       minor_breaks = seq(-5, -2, by = 0.5), 
                       labels = TeX(paste0(&quot;$10^{&quot;, -5:-2, &quot;}$&quot;))) +
    theme_bw() +
    theme(aspect.ratio = 1, 
          title = element_text(size = 10))

# CAT curve for BEER posterior probabilities
#   For a fair comparison, set super-enriched peptides in the replicate sample
#   to have a rank of 1 as all of these peptides are identified as enriched
#   by BEER and edgeR. 
se_peps &lt;- cs_tidy %&gt;% 
    filter(sample == cs_sample &amp; is_se) %&gt;%
    group_by(pair_id) %&gt;%
    filter(row_number() == 1) %&gt;%
    ungroup() %&gt;%
    mutate(rank = 1:n())

#   Ties are broken by the posterior marginal estimates of phi
beer_ranks &lt;- cs_tidy %&gt;%
    filter(sample == cs_sample) %&gt;%
    group_by(pair_num) %&gt;%
    arrange(desc(beer_prob), desc(beer_logfc), pair_id, .by_group = TRUE) %&gt;%
    mutate(rank_se = nrow(se_peps) + cumsum(!pair_id %in% se_peps$pair_id)) %&gt;%
    left_join(se_peps %&gt;% select(pair_id, rank), by = c(&quot;pair_id&quot;)) %&gt;%
    mutate(rank = ifelse(pair_id %in% se_peps$pair_id, rank, rank_se)) %&gt;%
    select(pair_id, pair_num, rank) %&gt;%
    pivot_wider(names_from = &quot;pair_num&quot;, 
                values_from = &quot;pair_id&quot;) %&gt;%
    dplyr::rename(pair_1 = `1`, pair_2 = `2`) %&gt;%
    arrange(rank)

beer_conc &lt;- sapply(1:nrow(beer_ranks), function(rank){
    length(intersect(beer_ranks$pair_1[beer_ranks$rank &lt;= rank], 
                     beer_ranks$pair_2[beer_ranks$rank &lt;= rank]))/rank
})

# CAT curve for edgeR p-values
edgeR_ranks &lt;- cs_tidy %&gt;%
    filter(sample == cs_sample) %&gt;%
    group_by(pair_num) %&gt;%
    arrange(edgeR_bh, desc(edgeR_logfc), pair_id, .by_group = TRUE) %&gt;%
    mutate(rank_se = nrow(se_peps) + cumsum(!pair_id %in% se_peps$pair_id)) %&gt;%
    left_join(se_peps %&gt;% select(pair_id, rank), by = c(&quot;pair_id&quot;)) %&gt;%
    mutate(rank = ifelse(pair_id %in% se_peps$pair_id, rank, rank_se)) %&gt;%
    select(pair_id, pair_num, rank) %&gt;%
    pivot_wider(names_from = &quot;pair_num&quot;, 
                values_from = &quot;pair_id&quot;) %&gt;%
    dplyr::rename(pair_1 = `1`, pair_2 = `2`) %&gt;%
    arrange(rank)

edgeR_conc &lt;- sapply(1:nrow(edgeR_ranks), function(rank){
    length(intersect(edgeR_ranks$pair_1[edgeR_ranks$rank &lt;= rank], 
                     edgeR_ranks$pair_2[edgeR_ranks$rank &lt;= rank]))/rank
})

cat_plot &lt;- data.frame(rank = 1:length(beer_conc), 
                       BEER = beer_conc, 
                       edgeR = edgeR_conc) %&gt;%
    pivot_longer(cols = c(&quot;BEER&quot;, &quot;edgeR&quot;), 
                 names_to = &quot;approach&quot;, 
                 values_to = &quot;concordance&quot;) %&gt;%
    ggplot(aes(x = rank, y = concordance, color = approach)) +
    geom_line() +
    labs(title = &quot;Concordance at the top&quot;, 
         y = &quot;Concordance&quot;, 
         x = &quot;Rank&quot;) +
    coord_cartesian(ylim = c(0, 1), xlim = c(0, 100)) +
    scale_x_continuous(breaks = seq(0, 100, by = 10)) +
    scale_y_continuous(breaks = seq(0, 1, by = 0.2)) +
    scale_color_manual(values = c(&quot;red&quot;, &quot;black&quot;)) +
    theme_bw() + 
    theme(aspect.ratio = 1, 
          title = element_text(size = 10),
          legend.background = element_rect(color = &quot;black&quot;, size = 0.3), 
          legend.position = c(0.8, 0.2))

ggarrange(prop_reads, cat_plot, nrow = 1, ncol = 2)</code></pre>
<p><img src="figure/coronascan.Rmd/figure_coronascan_replicates-1.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-s15" class="section level3">
<h3>Figure S15</h3>
<p>Coronascan peptide pairs by rank. For each set of unique peptides, peptides are sorted in decreasing order by posterior probability and -log10 edgeR p-values.</p>
<pre class="r"><code>post_prob &lt;- cs_tidy %&gt;%
    filter(sample == cs_sample) %&gt;%
    select(pair_num, beer_prob) %&gt;%
    group_by(pair_num) %&gt;%
    arrange(desc(beer_prob), .by_group = TRUE) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    ggplot(aes(x = rank, y = beer_prob, group = pair_num, color = pair_num)) + 
    geom_point(size = 1) +
    geom_line() +
    coord_cartesian(xlim = c(0, 100), ylim = c(0, 1)) +
    labs(title = &quot;BEER&quot;, 
         x = &quot;rank&quot;, 
         y = &quot;posterior probabilities&quot;, 
         color = &quot;peptide&quot;) +
    scale_color_manual(values = c(&quot;red&quot;, &quot;black&quot;)) +
    theme_bw() + 
    theme(aspect.ratio = 1, 
          title = element_text(size = 10), 
          legend.background = element_rect(color = &quot;black&quot;, size = 0.3), 
          legend.position = &quot;none&quot;)

p_value &lt;- cs_tidy %&gt;%
    filter(sample == cs_sample) %&gt;%
    select(pair_num, edgeR_prob) %&gt;%
    group_by(pair_num) %&gt;%
    arrange(desc(edgeR_prob), .by_group = TRUE) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    ggplot(aes(x = rank, y = edgeR_prob, group = pair_num, color = pair_num)) + 
    geom_point(size = 1) +
    geom_line() +
    labs(title = &quot;edgeR&quot;, 
         x = &quot;rank&quot;, 
         y = &quot;-log10(p-values)&quot;, 
         color = &quot;peptide&quot;) +
    coord_cartesian(xlim = c(0, 100), ylim = c(0, 6)) +
    scale_y_continuous(oob = scales::oob_censor_any) +
    scale_color_manual(values = c(&quot;red&quot;, &quot;black&quot;)) +
    theme_bw() +
    theme(aspect.ratio = 1, 
          title = element_text(size = 10), 
          legend.background = element_rect(color = &quot;black&quot;, size = 0.3), 
          legend.position = c(0.825, 0.825))

ggarrange(post_prob, p_value, nrow = 1)</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 2 row(s) containing missing values (geom_path).</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 1 row(s) containing missing values (geom_path).</code></pre>
<p><img src="figure/coronascan.Rmd/figure_coronascan_ranked_prob-1.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="tables" class="section level2">
<h2>Tables</h2>
<div id="table-s4" class="section level3">
<h3>Table S4</h3>
<p>Concordance of enrichment calls between peptide pairs for all CS samples. Each sample has 3366 unique peptide pairs. n: number of pairs with concordant enrichment calls; p: proportion of pairs with concordant enrichment calls.</p>
<pre class="r"><code>cs_tidy %&gt;%
    select(sample, pair_id, pair_num, beer_hits, edgeR_hits) %&gt;%
    pivot_longer(cols = contains(&quot;hits&quot;), 
                 names_to = &quot;method&quot;, 
                 values_to = &quot;hits&quot;, 
                 names_pattern = &quot;([A-Za-z]*)_hits&quot;) %&gt;%
    pivot_wider(names_from = pair_num, values_from = hits) %&gt;%
    mutate(concordant = ifelse(`1` == `2`, 1, 0))%&gt;%
    group_by(sample, method) %&gt;%
    summarize(num = sum(concordant), 
              total_pep = n(), 
              prop = num/total_pep, .groups = &quot;drop&quot;) %&gt;%
    pivot_longer(cols = c(&quot;num&quot;, &quot;prop&quot;), 
                 names_to = &quot;param&quot;, 
                 values_to = &quot;value&quot;) %&gt;%
    unite(&quot;method_param&quot;, method, param) %&gt;%
    pivot_wider(names_from = &quot;method_param&quot;, values_from = &quot;value&quot;) %&gt;%
    select(-total_pep) %&gt;%
    mutate(across(contains(c(&quot;beer&quot;, &quot;edgeR&quot;)), round, digits = 3)) %&gt;%
    kbl(col.names = c(&quot;sample&quot;, rep(c(&quot;n&quot;, &quot;p&quot;), times = 2))) %&gt;%
    kable_styling(
        bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;, &quot;responsive&quot;)) %&gt;%
    add_header_above(c(&quot; &quot; = 1, &quot;BEER&quot; = 2, &quot;edgeR&quot; = 2))</code></pre>
<table class="table table-striped table-hover table-condensed table-responsive" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="empty-cells: hide;border-bottom:hidden;" colspan="1">
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="2">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
BEER
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="2">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
edgeR
</div>
</th>
</tr>
<tr>
<th style="text-align:left;">
sample
</th>
<th style="text-align:right;">
n
</th>
<th style="text-align:right;">
p
</th>
<th style="text-align:right;">
n
</th>
<th style="text-align:right;">
p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
BEADS 1
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
BEADS 2
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
BEADS 3
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
BEADS 4
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
BEADS 5
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
BEADS 6
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
BEADS 7
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
BEADS 8
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
3466
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
CS 1
</td>
<td style="text-align:right;">
3452
</td>
<td style="text-align:right;">
0.996
</td>
<td style="text-align:right;">
3454
</td>
<td style="text-align:right;">
0.997
</td>
</tr>
<tr>
<td style="text-align:left;">
CS 2
</td>
<td style="text-align:right;">
3443
</td>
<td style="text-align:right;">
0.993
</td>
<td style="text-align:right;">
3443
</td>
<td style="text-align:right;">
0.993
</td>
</tr>
<tr>
<td style="text-align:left;">
CS 3
</td>
<td style="text-align:right;">
3430
</td>
<td style="text-align:right;">
0.990
</td>
<td style="text-align:right;">
3423
</td>
<td style="text-align:right;">
0.988
</td>
</tr>
<tr>
<td style="text-align:left;">
CS 4
</td>
<td style="text-align:right;">
3432
</td>
<td style="text-align:right;">
0.990
</td>
<td style="text-align:right;">
3427
</td>
<td style="text-align:right;">
0.989
</td>
</tr>
<tr>
<td style="text-align:left;">
CS 5
</td>
<td style="text-align:right;">
3445
</td>
<td style="text-align:right;">
0.994
</td>
<td style="text-align:right;">
3444
</td>
<td style="text-align:right;">
0.994
</td>
</tr>
<tr>
<td style="text-align:left;">
CS 6
</td>
<td style="text-align:right;">
3435
</td>
<td style="text-align:right;">
0.991
</td>
<td style="text-align:right;">
3433
</td>
<td style="text-align:right;">
0.990
</td>
</tr>
<tr>
<td style="text-align:left;">
CS 8
</td>
<td style="text-align:right;">
3455
</td>
<td style="text-align:right;">
0.997
</td>
<td style="text-align:right;">
3450
</td>
<td style="text-align:right;">
0.995
</td>
</tr>
<tr>
<td style="text-align:left;">
CS 7
</td>
<td style="text-align:right;">
3450
</td>
<td style="text-align:right;">
0.995
</td>
<td style="text-align:right;">
3446
</td>
<td style="text-align:right;">
0.994
</td>
</tr>
<tr>
<td style="text-align:left;">
CS 10
</td>
<td style="text-align:right;">
3441
</td>
<td style="text-align:right;">
0.993
</td>
<td style="text-align:right;">
3439
</td>
<td style="text-align:right;">
0.992
</td>
</tr>
<tr>
<td style="text-align:left;">
CS 9
</td>
<td style="text-align:right;">
3441
</td>
<td style="text-align:right;">
0.993
</td>
<td style="text-align:right;">
3437
</td>
<td style="text-align:right;">
0.992
</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] beer_0.99.0                 rjags_4-10                 
 [3] coda_0.19-4                 PhIPData_1.1.2             
 [5] SummarizedExperiment_1.23.4 Biobase_2.53.0             
 [7] GenomicRanges_1.45.0        GenomeInfoDb_1.29.4        
 [9] IRanges_2.27.2              S4Vectors_0.31.2           
[11] BiocGenerics_0.39.2         MatrixGenerics_1.5.4       
[13] matrixStats_0.60.1          BiocManager_1.30.16        
[15] RColorBrewer_1.1-2          kableExtra_1.3.4           
[17] latex2exp_0.5.0             gridExtra_2.3              
[19] ggpubr_0.4.0                here_1.0.1                 
[21] forcats_0.5.1               stringr_1.4.0              
[23] dplyr_1.0.7                 purrr_0.3.4                
[25] readr_2.0.1                 tidyr_1.1.3                
[27] tibble_3.1.4                ggplot2_3.3.5              
[29] tidyverse_1.3.1             plyr_1.8.6                 
[31] workflowr_1.6.2            

loaded via a namespace (and not attached):
 [1] colorspace_2.0-2       ggsignif_0.6.2         ellipsis_0.3.2        
 [4] rio_0.5.27             rprojroot_2.0.2        XVector_0.33.0        
 [7] fs_1.5.0               rstudioapi_0.13        farver_2.1.0          
[10] listenv_0.8.0          fansi_0.5.0            lubridate_1.7.10      
[13] xml2_1.3.2             codetools_0.2-18       knitr_1.33            
[16] jsonlite_1.7.2         broom_0.7.9            dbplyr_2.1.1          
[19] compiler_4.1.1         httr_1.4.2             backports_1.2.1       
[22] assertthat_0.2.1       Matrix_1.3-4           fastmap_1.1.0         
[25] limma_3.49.4           cli_3.0.1              later_1.3.0           
[28] htmltools_0.5.2        tools_4.1.1            gtable_0.3.0          
[31] glue_1.4.2             GenomeInfoDbData_1.2.6 Rcpp_1.0.7            
[34] carData_3.0-4          cellranger_1.1.0       vctrs_0.3.8           
[37] svglite_2.0.0          progressr_0.8.0        xfun_0.25             
[40] globals_0.14.0         openxlsx_4.2.4         rvest_1.0.1           
[43] lifecycle_1.0.0        rstatix_0.7.0          future_1.22.1         
[46] edgeR_3.35.0           zlibbioc_1.39.0        scales_1.1.1          
[49] hms_1.1.0              promises_1.2.0.1       parallel_4.1.1        
[52] yaml_2.2.1             curl_4.3.2             stringi_1.7.4         
[55] highr_0.9              zip_2.2.0              rlang_0.4.11          
[58] pkgconfig_2.0.3        systemfonts_1.0.2      bitops_1.0-7          
[61] evaluate_0.14          lattice_0.20-44        labeling_0.4.2        
[64] cowplot_1.1.1          tidyselect_1.1.1       parallelly_1.27.0     
[67] magrittr_2.0.1         R6_2.5.1               generics_0.1.0        
[70] DelayedArray_0.19.1    DBI_1.1.1              pillar_1.6.2          
[73] haven_2.4.3            whisker_0.4            foreign_0.8-81        
[76] withr_2.4.2            abind_1.4-5            RCurl_1.98-1.4        
[79] future.apply_1.8.1     modelr_0.1.8           crayon_1.4.1          
[82] car_3.0-11             utf8_1.2.2             tzdb_0.1.2            
[85] rmarkdown_2.10         locfit_1.5-9.4         grid_4.1.1            
[88] readxl_1.3.1           data.table_1.14.0      git2r_0.28.0          
[91] reprex_2.0.1           digest_0.6.27          webshot_0.5.2         
[94] httpuv_1.6.2           munsell_0.5.0          viridisLite_0.4.0     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
