<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>HIV Discussion</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />
<link rel="apple-touch-icon" sizes="180x180" href="site_libs/images/beer_icons/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="site_libs/images/beer_icons/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="site_libs/images/beer_icons/favicon-16x16.png">
<link rel="manifest" href="site_libs/images/beer_icons/site.webmanifest">
<link rel="mask-icon" href="site_libs/images/beer_icons/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#b91d47">
<meta name="theme-color" content="#ffffff">

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">beer_manuscript</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="beer.html">BEER</a>
</li>
<li>
  <a href="simulation.html">Simulated Data</a>
</li>
<li>
  <a href="hiv_ec.html">HIV Elite Controllers</a>
</li>
<li>
  <a href="coronascan.html">CoronaScan Samples</a>
</li>
<li>
  <a href="hiv_discussion.html">HIV Discussion</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="templink">
    <span class="fa fa-file-alt"></span>
     
    Manuscript
  </a>
</li>
<li>
  <a href="https://github.com/athchen/beer_manuscript">
    <span class="fa fa-github"></span>
     
    Source Code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">HIV Discussion</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>beer_manuscript/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210907code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210907)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210907code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210907)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomathchenbeermanuscripttree924ae62e2b84adb934a49370ea92576b8196811atargetblank924ae62a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/athchen/beer_manuscript/tree/924ae62e2b84adb934a49370ea92576b8196811a" target="_blank">924ae62</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomathchenbeermanuscripttree924ae62e2b84adb934a49370ea92576b8196811atargetblank924ae62a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/athchen/beer_manuscript/tree/924ae62e2b84adb934a49370ea92576b8196811a" target="_blank">924ae62</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/partials/.DS_Store
    Ignored:    data_processed/.DS_Store
    Ignored:    data_processed/simulation_2beads_mle/.DS_Store
    Ignored:    data_processed/simulation_2beads_mom/.DS_Store
    Ignored:    data_processed/simulation_2beads_truth/.DS_Store
    Ignored:    data_processed/simulation_4beads_edgeR/.DS_Store
    Ignored:    data_processed/simulation_curves.rda
    Ignored:    data_raw/.DS_Store
    Ignored:    figures/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/hiv_discussion.Rmd</code>) and HTML (<code>docs/hiv_discussion.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/d8b023fb502b200a28b64e20a473e1d1520a2042/docs/hiv_discussion.html" target="_blank">d8b023f</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/athchen/beer_manuscript/75f2a6d707f4b9253cdf10d8b7faa34ced412008/docs/hiv_discussion.html" target="_blank">75f2a6d</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/athchen/beer_manuscript/blob/9d7b43ac86071aa909acd9f4d9807d3219c15b7e/analysis/hiv_discussion.Rmd" target="_blank">9d7b43a</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-18
</td>
<td>
Updated repo to use most up-to-date figures and code
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="additional-hiv-data" class="section level2">
<h2>Additional HIV Data</h2>
<p>This data was used to highlight strong peptide-dependent binding in the Discussion and had been previously analyzed to characterize antibody responses and their evolution during the course of HIV infection (<a href="https://doi.org/10.1016/j.celrep.2019.03.097">Eshleman et. al 2019</a>) and to generate a classifier for recent HIV infection (<a href="https://doi.org/10.1002/sim.8920">Chen et. al 2021</a>).</p>
<pre class="r"><code>#&#39; Code to load required packages to reproduce the results and figures in 
#&#39; the manuscript. 
required_packages &lt;- c(&#39;plyr&#39;, &#39;tidyverse&#39;, &#39;here&#39;, &#39;ggpubr&#39;, &#39;gridExtra&#39;, 
                       &#39;latex2exp&#39;, &#39;kableExtra&#39;, &#39;RColorBrewer&#39;, &#39;BiocManager&#39;)
for (pkg in required_packages) {
    if (!(pkg %in% rownames(installed.packages()))) {
        install.packages(pkg)
    }
    library(pkg, character.only = TRUE)
}

bioc_packages &lt;- c(&quot;beer&quot;)
for(pkg in bioc_packages){
    if(!(pkg %in% rownames(installed.packages()))) {
        BiocManager::install(pkg)
    }
    library(pkg, character.only = TRUE)
}

rm(list = c(&quot;required_packages&quot;, &quot;bioc_packages&quot;, &quot;pkg&quot;))

#&#39; Define global variables for plotting
hot_cold_cols &lt;- c(&quot;navy&quot;, &quot;blue&quot;, &quot;deepskyblue&quot;, &quot;cyan&quot;, &quot;lightcyan&quot;,
                   &quot;yellow&quot;, &quot;orange&quot;, &quot;red&quot;)</code></pre>
<pre class="r"><code>#&#39; helper_functions.R 
#&#39; 
#&#39; functions used to help process and analyze output from BEER simulations

#&#39; integrate_vector()
#&#39; 
#&#39; Function to perform trapezoidal approximation given two vectors. 
#&#39; 
#&#39; @param x numeric vector of x values
#&#39; @param y numeric vector with the same length as x
#&#39; 
#&#39; @return numeric value of the trapezoidal approximation 
integrate_vector &lt;- function(x, y){
    
    if (length(y) != length(x)) {
        stop(&quot;The length of the vectors must be equal.&quot;)
    }
    
    n_points &lt;- length(y)
    sorted_ind &lt;- sort(x, index.return = TRUE)$ix
    sorted_x &lt;- x[sorted_ind]
    sorted_y &lt;- y[sorted_ind]
    
    # trapezoidal rule
    sum(0.5 * (sorted_y[-1] + sorted_y[-n_points]) * 
            (sorted_x[-1] - sorted_x[-n_points]))
}

#&#39; get_roc()
#&#39; 
#&#39; Function to calculate the ppv, sens, and spec for various
#&#39; cutoffs between 0 and 1. 
#&#39; 
#&#39; @param data data frame  with columns `prop_enriched` for the threshold and 
#&#39; `Z` for the true enrichment status, and optional column `extra_info`
#&#39; @param min_cutoff minimum cutoff, default to 0
#&#39; @param max_cutoff maximum cutoff, default ot 1 - 1e-6
#&#39; @param extra_info boolean indicating whether extra information should be used
#&#39; in the classification at each cutoff. If `TRUE`, then the column `extra_info`
#&#39; must be present in the data frame. 
#&#39; 
#&#39; @return data fram with columns `cutoff`, `ppv`, `sens`, and `spec`. 
get_roc &lt;- function(data, min_cutoff = 0, max_cutoff = 1 - 1e-6, 
                    extra_info = FALSE){
    
    cutoffs &lt;- seq(min_cutoff, max_cutoff, length.out = 1000)
    
    # Calculate ppv, sens, spec for each cutoff
    ppv &lt;- sapply(cutoffs, function(x) {
        if(extra_info){
            predict &lt;- (data$prop_enriched &gt;= x &amp; data$extra_info)
        } else {
            predict &lt;- (data$prop_enriched &gt;= x)
        }
        sum((data$Z == 1) &amp; predict)/sum(predict)
    })
    
    spec &lt;- sapply(cutoffs, function(x){
        if(extra_info){
            predict &lt;- (data$prop_enriched &gt;= x &amp; data$extra_info)
        } else {
            predict &lt;- (data$prop_enriched &gt;= x)
        }
        sum((data$Z == 0) &amp; !predict)/sum(data$Z == 0)
    })
    
    sens &lt;- sapply(cutoffs, function(x){
        if(extra_info){
            predict &lt;- (data$prop_enriched &gt;= x &amp; data$extra_info)
        } else {
            predict &lt;- (data$prop_enriched &gt;= x)
        }
        sum((data$Z == 1) &amp; predict)/sum(data$Z == 1)
    })
    
    # Get AUC for ROC
    npoints &lt;- length(sens)
    area_roc &lt;- integrate_vector(1-spec, sens)
    
    return(cutoffs = data.frame(cutoff = cutoffs,
                                ppv = ppv,
                                sens = sens,
                                spec = spec, 
                                area_roc = area_roc))
}

#&#39; get_legend()
#&#39; 
#&#39; Function given [here](https://stackoverflow.com/questions/12539348/ggplot-separate-legend-and-plot)
#&#39; to extract the legend for plotting purposes. 
#&#39; 
#&#39; @param plot ggplot
#&#39; @return ggplot legend
get_legend &lt;- function(myggplot){
    tmp &lt;- ggplot_gtable(ggplot_build(myggplot))
    leg &lt;- which(sapply(tmp$grobs, function(x) x$name) == &quot;guide-box&quot;)
    legend &lt;- tmp$grobs[[leg]]
    return(legend)
}

#&#39; mysqrt_trans()
#&#39; 
#&#39; Function given [here](https://stackoverflow.com/questions/47944992/ggplot2-removes-zero-when-using-scale-x-sqrt)
#&#39; to add zero to the plot after sqrt transforming the x-axis
#&#39; 
#&#39; @param plot ggplot
#&#39; @return ggplot legend
mysqrt_trans &lt;- function() {
    scales::trans_new(&quot;mysqrt&quot;, 
                      transform = base::sqrt,
                      inverse = function(x) ifelse(x&lt;0, 0, x^2),
                      domain = c(0, Inf))
}

#&#39; penriched_fit()
#&#39; 
#&#39; Function that returns a data frame with the point estimate, and 95\% 
#&#39; confidence intervals. 
#&#39; 
#&#39; @param model logistic regression model
#&#39; @param covariates data frame of covariates
#&#39; @return data frame with columns for the covariate, point estimate, lower CI
#&#39; and upper CI.
penriched_fit &lt;- function(model, covariates){
    
    # Predict based on the model
    prediction &lt;- predict(model, covariates, type = &quot;link&quot;, se.fit = TRUE)
    pred_lower &lt;- prediction$fit - 1.96*prediction$se.fit
    pred_upper &lt;- prediction$fit + 1.96*prediction$se.fit
    
    # Transform logit to probabilities
    ppred &lt;- 1/(1 + exp(-prediction$fit))
    plower &lt;- 1/(1 + exp(-pred_lower))
    pupper &lt;- 1/(1 + exp(-pred_upper))
    
    # Return covariates with prediction + 95 CI added
    bind_cols(covariates, 
              data.frame(predict_p = ppred, 
                         lower_ci = plower, 
                         upper_ci = pupper))
}</code></pre>
<pre class="r"><code>hiv_virscan &lt;- readRDS(here::here(&quot;data_raw&quot;, &quot;hiv_virscan.rds&quot;))
hiv_beads &lt;- hiv_virscan[, hiv_virscan$group == &quot;beads&quot;]</code></pre>
</div>
<div id="figures" class="section level2">
<h2>Figures</h2>
<div id="figure-s13" class="section level3">
<h3>Figure S13</h3>
<p>Evidence for a strong peptide effect in PhIP-Seq data, demonstrated using data from five plates of a previous experiment using HIV samples, analyzed in <a href="https://doi.org/10.1016/j.celrep.2019.03.097">Eshleman et. al 2019</a> and <a href="https://doi.org/10.1002/sim.8920">Chen et. al 2021</a>. <strong>Left:</strong> observed read counts per million reads for 95,242 peptides from two “beads only” samples from different plates. For these two samples, the Spearman correlation is 0.875. <strong>Middle:</strong> observed average read counts per million reads for 95,242 peptides from all “beads only” samples from two different plates. For these two plates, the Spearman correlation is 0.975. <strong>Right:</strong> within and between plate sample correlations as a function of sequencing depth. For each pair of bead only samples from the same plate (red dots, 117 pairs total), the Spearman correlation (y-axis) is related to the minimum of the respective two sequencing depths (x-axis). For each pair of plates (blue dots, 10 pairs total), the Spearman correlation between the average read counts of the bead only samples (y-axis) is also related to the minimum of the two median sequencing depths (x-axis), and substantially higher than the correlations of the bead only samples.</p>
<pre class="r"><code>## Left panel: peptide effect in beads-only samples
cpm &lt;- propReads(hiv_beads)*1e6
set.seed(1)
breaks &lt;- c(1, 2, 5, 10, 20, 50, 100)

plot_pepeff &lt;- data.frame(beads_s1p3 = cpm[, &quot;beads_3_C9&quot;], 
                          beads_s1p4 = cpm[, &quot;beads_4_A5&quot;]) %&gt;%
    ggplot(aes(x = log10(beads_s1p3 + 1), 
               y = log10(beads_s1p4 + 1))) + 
    geom_jitter(width = 0.01, height = 0.01, size = 0.1) +
    geom_abline(aes(slope = 1, intercept = 0), color = &quot;red&quot;) +
    scale_x_continuous(breaks = log10(breaks), 
                       labels = breaks) +
    scale_y_continuous(breaks = log10(breaks), 
                       labels = breaks) +
    coord_fixed(xlim = c(0, log10(100)), ylim = c(0, log10(100))) +
    labs(x = &quot;reads per million + 1 [sample 1 plate 3]&quot;, 
         y = &quot;reads per million + 1 [sample 1 plate 4]&quot;) +
    theme_bw() + 
    theme(panel.grid = element_blank(), 
          aspect.ratio = 1)

# # Spearman correlation 
# cor(cpm[, &quot;beads_3_C9&quot;], cpm[, &quot;beads_4_A5&quot;], 
#     method = &quot;spearman&quot;)

## Middle panel: average cpm
avg_cpm &lt;- vapply(paste0(&quot;plate&quot;, 1:5), function(plate){
    plate_num &lt;- str_match(plate, &quot;plate([0-9])&quot;)[, 2]
    rowMeans(cpm[, grepl(paste0(&quot;beads_&quot;, plate_num), colnames(cpm))])
}, numeric(nrow(cpm)))

plot_avgcpm &lt;- avg_cpm %&gt;%
    as_tibble() %&gt;%
    ggplot(aes(x = log10(plate2 + 1), y = log10(plate3 + 1))) + 
    geom_jitter(width = 0.01, height = 0.01, size = 0.1) +
    geom_abline(aes(slope = 1, intercept = 0), color = &quot;red&quot;) +
    scale_x_continuous(breaks = log10(breaks), 
                       labels = breaks) +
    scale_y_continuous(breaks = log10(breaks), 
                       labels = breaks) +
    coord_fixed(xlim = c(0, log10(100)), ylim = c(0, log10(100))) +
    labs(x = &quot;average reads per million + 1 [plate 2]&quot;, 
         y = &quot;average reads per million + 1 [plate 3]&quot;) +
    theme_bw() + 
    theme(panel.grid = element_blank(), 
          aspect.ratio = 1)

# # Spearman correlation
# cor(avg_cpm[, &quot;plate2&quot;], avg_cpm[, &quot;plate3&quot;], method = &quot;spearman&quot;)   

## Right panel: correlation by read depth
# correlations between samples within plates
within_pl &lt;- cor(cpm, method = &quot;spearman&quot;) 
within_pl[lower.tri(within_pl, diag = TRUE)] &lt;- NA
within_df &lt;- within_pl %&gt;%
    as_tibble(rownames = &quot;sample_1&quot;) %&gt;%
    pivot_longer(cols = -sample_1, 
                 names_to = &quot;sample_2&quot;, 
                 values_to = &quot;corr&quot;) %&gt;%
    mutate(pl_1 = str_match(sample_1, &quot;beads_([0-9])&quot;)[, 2], 
           pl_2 = str_match(sample_2, &quot;beads_([0-9])&quot;)[, 2]) %&gt;%
    left_join(data.frame(sample_id = colnames(hiv_beads), 
                         rd_1 = librarySize(hiv_beads)),
              by = c(&quot;sample_1&quot;= &quot;sample_id&quot;)) %&gt;%
    left_join(data.frame(sample_id = colnames(hiv_beads), 
                         rd_2 = librarySize(hiv_beads)),
              by = c(&quot;sample_2&quot;= &quot;sample_id&quot;)) %&gt;%
    filter(!is.na(corr) &amp; pl_1 == pl_2) %&gt;%
    rowwise() %&gt;%
    mutate(min_rd = min(rd_1, rd_2)) %&gt;%
    ungroup()

# correlations between plate averages
med_rd &lt;- vapply(paste0(&quot;plate&quot;, 1:5), function(plate){
    plate_num &lt;- str_match(plate, &quot;plate([0-9])&quot;)[, 2]
    median(librarySize(hiv_beads)[grepl(paste0(&quot;beads_&quot;, plate_num), colnames(cpm))])
}, numeric(1))

between_pl &lt;- cor(avg_cpm, method = &quot;spearman&quot;)
between_pl[lower.tri(between_pl, diag = TRUE)] &lt;- NA
between_df &lt;- between_pl %&gt;%
    as_tibble(rownames = &quot;plate_1&quot;) %&gt;%
    pivot_longer(cols = -plate_1, 
                 names_to = &quot;plate_2&quot;, 
                 values_to = &quot;corr&quot;) %&gt;%
    left_join(data.frame(plate = names(med_rd), 
                         med_rd_1 = med_rd),
              by = c(&quot;plate_1&quot;= &quot;plate&quot;)) %&gt;%
    left_join(data.frame(plate = names(med_rd), 
                         med_rd_2 = med_rd),
              by = c(&quot;plate_2&quot;= &quot;plate&quot;)) %&gt;%
    filter(!is.na(corr)) %&gt;%
    rowwise() %&gt;%
    mutate(med_rd = min(med_rd_1, med_rd_2)) %&gt;%
    ungroup() 

plot_corr &lt;- ggplot() +
    geom_jitter(aes(x = min_rd/1e6, y = corr), 
                data = within_df,  width = 0.005, color = &quot;red&quot;) +
    geom_jitter(aes(x = med_rd/1e6, y = corr),
               data = between_df, width = 0.005, color = &quot;blue&quot;) +
    scale_x_continuous(breaks = seq(1.5, 4.0, by = 0.5)) +
    scale_y_continuous(breaks = seq(0.5, 1.0, by = 0.1)) +
    coord_cartesian(xlim = c(1.5, 4.0), ylim = c(0.5, 1.0)) +
    labs(x = &quot;sample minimum read depth [million]&quot;, 
         y = &quot;spearman correlation&quot;) +
    theme_bw() + 
    theme(aspect.ratio = 1)

ggarrange(plot_pepeff, plot_avgcpm, plot_corr, ncol = 3, 
          align = &quot;v&quot;)    </code></pre>
<p><img src="figure/hiv_discussion.Rmd/figure_corstructure-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-figure_corstructure-1">
Past versions of figure_corstructure-1.png
</button>
</p>
<div id="fig-figure_corstructure-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/athchen/beer_manuscript/blob/75f2a6d707f4b9253cdf10d8b7faa34ced412008/docs/figure/hiv_discussion.Rmd/figure_corstructure-1.png" target="_blank">75f2a6d</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="figure-s14" class="section level3">
<h3>Figure S14</h3>
<p>Evidence for larger than binomial variability in the PhIP-Seq data analyzed in <a href="https://doi.org/10.1016/j.celrep.2019.03.097">Eshleman et. al 2019</a> and <a href="https://doi.org/10.1002/sim.8920">Chen et. al 2021</a>. <strong>Left:</strong> library size (reads in millions) for 36 control (“bead only”) samples from 5 plates. <strong>Middle:</strong> expected read counts per million reads aligned based on the estimated Binomial probabilities (dots, colored by plate) and respective 95% confidence intervals for a peptide with large expected read counts, for each of the control samples. Highlighted are samples 3 and 6 from plate 1, showing large discrepancies between the Binomial probabilities for this peptide between the two bead only samples. <strong>Right:</strong> the same statistics as in the middle panel, for a peptide with smaller expected read counts. Highlighted are samples 10 and 12 from plate 2, again showing a large discrepancy between the Binomial probabilities for this peptide between the two bead only samples.</p>
<pre class="r"><code># Plot library size
lib_size &lt;- data.frame(libsize = librarySize(hiv_beads)) %&gt;%
    rownames_to_column(&quot;sample&quot;) %&gt;%
    mutate(plate = str_match(sample, &quot;beads_([0-9])_&quot;)[, 2], 
           plate = factor(plate, levels = 1:5)) %&gt;%
    arrange(plate, sample) %&gt;%
    ggplot(aes(x = sample, y = libsize, fill = plate)) + 
    geom_bar(stat = &quot;identity&quot;, color = &quot;black&quot;, show.legend = FALSE) + 
    geom_point(shape = 21, size = 2, color = &quot;black&quot;, 
               data = data.frame(sample = colnames(hiv_beads), 
                                 plate = factor(hiv_beads$plate, levels = 1:5), 
                                 libsize = rep(NA_integer_, ncol(hiv_beads)))) +
    scale_fill_manual(values = brewer.pal(5,&quot;Accent&quot;), 
                      labels = paste(&quot;Plate&quot;, 1:5)) +
    scale_y_continuous(breaks = (0:4)*1e6, 
                       labels = 0:4, 
                       expand = expansion(mult = c(0, 0.02))) +
    scale_x_discrete(labels = NULL, 
                     expand = expansion(add = c(1, 0))) +
    labs(y = &quot;reads [ millions ]&quot;, 
         x = &quot;&quot;) +
    guides(fill = guide_legend(keywidth = 0.1, keyheight = 0.75)) +
    theme_classic() +
    theme(aspect.ratio = 1, 
          axis.ticks.x = element_blank(), 
          panel.grid = element_blank(), 
          legend.title = element_blank(), 
          legend.position = c(0.92, 0.85))

# Plot variability for a peptide with large expected read counts
pep_large &lt;- &quot;pep_10160&quot;
binom_large &lt;- data.frame(counts = as.vector(counts(hiv_beads[pep_large, ])), 
                          n = librarySize(hiv_beads)) %&gt;%
    rownames_to_column(&quot;sample_id&quot;) %&gt;%
    rowwise() %&gt;%
    mutate(cpm = counts/n*1e6,
           cpm_025 = binom.test(counts, n)$conf.int[1]*1e6, 
           cpm_975 = binom.test(counts, n)$conf.int[2]*1e6) %&gt;%
    ungroup() %&gt;%
    mutate(plate = str_match(sample_id, &quot;beads_([0-9])&quot;)[, 2], 
           plate = factor(plate, levels = 1:5), 
           size = ifelse(sample_id %in% c(&quot;beads_1_D12&quot;, &quot;beads_1_E8&quot;), 2, 1)) %&gt;%
    mutate(sample_id = factor(sample_id)) %&gt;%
    ggplot(aes(x = as.numeric(sample_id))) +
    geom_linerange(aes(ymin = cpm_025, ymax = cpm_975)) +
    geom_point(aes(y = cpm, size = size, fill = plate), shape = 21) +
    scale_fill_manual(values = brewer.pal(5,&quot;Accent&quot;), 
                      labels = paste(&quot;Plate&quot;, 1:5)) +
    scale_y_continuous(breaks = c(40, 60, 80, 100), 
                       limits = c(30, 110)) +
    scale_x_continuous(labels = NULL, 
                       breaks =  cumsum(table(hiv_beads$plate)[-5]) + 0.5,
                       sec.axis = dup_axis(), 
                       expand = expansion(add = c(1, 1))) +
    geom_text(aes(x = xpos, y = ypos, label = label), 
              hjust = 0.5, nudge_x = 0.5, vjust = -0.3, size = 3, 
              data = data.frame(xpos = c(4, 12, 19, 24.5, 31.5), 
                                ypos = rep(110, 5), 
                                label = paste(&quot;Plate&quot;, 1:5))) +
    coord_cartesian(ylim = c(30, 110)) +
    scale_size(breaks = c(1, 2), 
               range = c(1.5, 3)) +
    labs(y = &quot;expected reads per million&quot;, 
         x = &quot;&quot;) +
    theme_bw() +
    theme(
        aspect.ratio = 1,
        panel.grid.major = element_line(linetype = &quot;dotted&quot;, color = &quot;lightgrey&quot;),
        panel.grid.minor.y = element_line(linetype = &quot;dotted&quot;, color = &quot;lightgrey&quot;), 
        panel.grid.minor.x = element_blank(), 
        axis.ticks.length.x = unit(-11, &quot;pt&quot;),
        legend.position = &quot;none&quot;, 
        plot.margin = margin(0, 0, 0, 0)
    )

# Plot variability for a peptide with smaller expected read counts
pep_small &lt;- &quot;pep_58858&quot;
binom_small &lt;- data.frame(counts = as.vector(counts(hiv_beads[pep_small, ])), 
                          n = librarySize(hiv_beads)) %&gt;%
    rownames_to_column(&quot;sample_id&quot;) %&gt;%
    rowwise() %&gt;%
    mutate(cpm = counts/n*1e6,
           cpm_025 = binom.test(counts, n)$conf.int[1]*1e6, 
           cpm_975 = binom.test(counts, n)$conf.int[2]*1e6) %&gt;%
    ungroup() %&gt;%
    mutate(plate = str_match(sample_id, &quot;beads_([0-9])&quot;)[, 2], 
           plate = factor(plate, levels = 1:5), 
           size = ifelse(sample_id %in% c(&quot;beads_2_B5&quot;, &quot;beads_2_D5&quot;), 2, 1)) %&gt;%
    mutate(sample_id = factor(sample_id)) %&gt;%
    ggplot(aes(x = as.numeric(sample_id))) +
    geom_linerange(aes(ymin = cpm_025, ymax = cpm_975)) +
    geom_point(aes(y = cpm, size = size, fill = plate), shape = 21) +
    scale_fill_manual(values = brewer.pal(5,&quot;Accent&quot;), 
                      labels = paste(&quot;Plate&quot;, 1:5)) +
    scale_y_continuous(breaks = seq(0, 20, 5), 
                       limits = c(0, 20)) +
    scale_x_continuous(labels = NULL, 
                       breaks =  cumsum(table(hiv_beads$plate)[-5]) + 0.5,
                       sec.axis = dup_axis(), 
                       expand = expansion(add = c(1, 1))) +
    geom_text(aes(x = xpos, y = ypos, label = label), 
              hjust = 0.5, nudge_x = 0.5, vjust = -0.3, size = 3, 
              data = data.frame(xpos = c(4, 12, 19, 24.5, 31.5), 
                                ypos = rep(20, 5), 
                                label = paste(&quot;Plate&quot;, 1:5))) +
    coord_cartesian(ylim = c(0, 20)) +
    scale_size(breaks = c(1, 2), 
               range = c(1.5, 3)) +
    labs(y = &quot;expected reads per million&quot;, 
         x = &quot;&quot;) +
    theme_bw() +
    theme(
        aspect.ratio = 1,
        panel.grid.major = element_line(linetype = &quot;dotted&quot;, color = &quot;lightgrey&quot;),
        panel.grid.minor.y = element_line(linetype = &quot;dotted&quot;, color = &quot;lightgrey&quot;), 
        panel.grid.minor.x = element_blank(), 
        axis.ticks.length.x = unit(-11, &quot;pt&quot;),
        legend.position = &quot;none&quot;, 
        plot.margin = margin(0, 0, 0, 0)
    )


ggarrange(lib_size, binom_large, binom_small, nrow = 1, 
          align = &quot;v&quot;)</code></pre>
<pre><code>Warning: Removed 36 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/hiv_discussion.Rmd/figdatastructurebinom-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-figdatastructurebinom-1">
Past versions of figdatastructurebinom-1.png
</button>
</p>
<div id="fig-figdatastructurebinom-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/athchen/beer_manuscript/blob/75f2a6d707f4b9253cdf10d8b7faa34ced412008/docs/figure/hiv_discussion.Rmd/figdatastructurebinom-1.png" target="_blank">75f2a6d</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="figure-s15" class="section level3">
<h3>Figure S15</h3>
<p>Expected versus observed read counts for 95,242 peptides from a randomly selected serum sample. Expected read counts for each peptide were derived using maximum likelihood estimates from the negative controls on the same plate. Each point represents one peptide from one sample, and peptides were considered enriched (red) if the observed read count was over 2 times (left) and 5 times (right) the expected number of reads. Linear regression lines (blue) were fitted using the non-enriched peptides and compared to the line where observed and expected reads are equal (black). The observed counts were truncated at 250 observed reads to enhance the display.</p>
<pre class="r"><code># Convert to tidy format at add expected prop/counts
hiv_tidy &lt;- as(hiv_virscan[, hiv_virscan$plate == 3], &quot;DataFrame&quot;) %&gt;%
    as_tibble() %&gt;%
    group_by(sample) %&gt;%
    mutate(n = sum(counts)) %&gt;%
    ungroup() %&gt;%
    select(sample, group, n, peptide, counts) %&gt;%
    mutate(prop_reads = counts/n) %&gt;%
    group_by(peptide) %&gt;%
    mutate(expected_prop = mean(prop_reads[group == &quot;beads&quot;]), 
           expected_rc = expected_prop*n, 
           hit_2x = ifelse(counts &gt; 2*expected_rc, TRUE, FALSE), 
           hit_5x = ifelse(counts &gt; 5*expected_rc, TRUE, FALSE)) %&gt;%
    ungroup()

## Identify one random sample on plate 3 for visualization purposes
set.seed(123)
sample_id &lt;- sample(colnames(hiv_virscan[, hiv_virscan$group != &quot;beads&quot; &amp; hiv_virscan$plate == 3]), 1)

# Figure S15: attnconstant.png ----------
hiv_tidy %&gt;%
    filter(counts &lt;= 250 &amp; expected_rc &lt;= 250 &amp; sample == sample_id) %&gt;%
    pivot_longer(cols = contains(&quot;hit_&quot;), 
                 names_to = &quot;threshold&quot;, 
                 values_to = &quot;hit&quot;, 
                 names_pattern = &quot;hit_([0-9])x&quot;) %&gt;%
    ggplot(aes(x = expected_rc, y = counts, group = threshold)) + 
    facet_wrap(.~threshold) +
    geom_point(aes(color = hit), size = 0.5, alpha = 0.75) + 
    geom_abline(aes(intercept = 0, slope = 1), color = &quot;black&quot;, size = 0.5) + 
    geom_smooth(formula = y ~ x - 1,
                data = hiv_tidy %&gt;% 
                    filter(sample == sample_id) %&gt;%
                    pivot_longer(cols = contains(&quot;hit_&quot;), 
                                 names_to = &quot;threshold&quot;, 
                                 values_to = &quot;hit&quot;, 
                                 names_pattern = &quot;hit_([0-9])x&quot;) %&gt;%
                    filter(!hit),
                color = &quot;blue&quot;, method = &quot;lm&quot;, size = 0.5) +
    geom_text(aes(label = label), 
              x = 235, y = 10, size = 6, 
              #x = 57.75, y = 244, size = 5,
              data = data.frame(threshold = c(&quot;2&quot;, &quot;5&quot;), 
                                label = c(&quot;2X&quot;, &quot;5X&quot;))) +
    labs(x = &quot;Expected # of reads&quot;, 
         y = &quot;Observed # of reads&quot;, 
         color = &quot;&quot;) +
    scale_color_manual(breaks = c(&quot;TRUE&quot;, &quot;FALSE&quot;), 
                       values = c(&quot;red&quot;, &quot;grey70&quot;), 
                       labels = c(&quot;enriched&quot;, &quot;not enriched&quot;)) +
    coord_cartesian(xlim = c(0, 250), ylim = c(0, 250)) + 
    theme_bw() + 
    theme(aspect.ratio = 1, 
          strip.background = element_blank(), 
          strip.text = element_blank(),
          legend.position = &quot;top&quot;, 
          legend.box.margin=margin(-10,-10,-10,-10))</code></pre>
<p><img src="figure/hiv_discussion.Rmd/figure_attnconstant-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-figure_attnconstant-1">
Past versions of figure_attnconstant-1.png
</button>
</p>
<div id="fig-figure_attnconstant-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/athchen/beer_manuscript/blob/75f2a6d707f4b9253cdf10d8b7faa34ced412008/docs/figure/hiv_discussion.Rmd/figure_attnconstant-1.png" target="_blank">75f2a6d</a>
</td>
<td>
Athena Chen
</td>
<td>
2022-01-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] beer_0.99.0                 rjags_4-12                 
 [3] coda_0.19-4                 PhIPData_1.2.0             
 [5] SummarizedExperiment_1.24.0 Biobase_2.54.0             
 [7] GenomicRanges_1.46.1        GenomeInfoDb_1.30.0        
 [9] IRanges_2.28.0              S4Vectors_0.32.3           
[11] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
[13] matrixStats_0.61.0          BiocManager_1.30.16        
[15] RColorBrewer_1.1-2          kableExtra_1.3.4           
[17] latex2exp_0.5.0             gridExtra_2.3              
[19] ggpubr_0.4.0                here_1.0.1                 
[21] forcats_0.5.1               stringr_1.4.0              
[23] dplyr_1.0.7                 purrr_0.3.4                
[25] readr_2.1.1                 tidyr_1.1.4                
[27] tibble_3.1.6                ggplot2_3.3.5              
[29] tidyverse_1.3.1             plyr_1.8.6                 
[31] workflowr_1.7.0            

loaded via a namespace (and not attached):
 [1] colorspace_2.0-2       ggsignif_0.6.3         ellipsis_0.3.2        
 [4] rprojroot_2.0.2        XVector_0.34.0         fs_1.5.2              
 [7] rstudioapi_0.13        farver_2.1.0           listenv_0.8.0         
[10] fansi_1.0.2            lubridate_1.8.0        xml2_1.3.3            
[13] splines_4.1.2          codetools_0.2-18       knitr_1.37            
[16] jsonlite_1.7.3         broom_0.7.11           dbplyr_2.1.1          
[19] compiler_4.1.2         httr_1.4.2             backports_1.4.1       
[22] assertthat_0.2.1       Matrix_1.4-0           fastmap_1.1.0         
[25] limma_3.50.0           cli_3.1.1              later_1.3.0           
[28] htmltools_0.5.2        tools_4.1.2            gtable_0.3.0          
[31] glue_1.6.0             GenomeInfoDbData_1.2.7 Rcpp_1.0.8            
[34] carData_3.0-5          cellranger_1.1.0       jquerylib_0.1.4       
[37] vctrs_0.3.8            nlme_3.1-155           svglite_2.0.0         
[40] progressr_0.10.0       xfun_0.29              globals_0.14.0        
[43] ps_1.6.0               rvest_1.0.2            lifecycle_1.0.1       
[46] rstatix_0.7.0          future_1.23.0          edgeR_3.36.0          
[49] getPass_0.2-2          zlibbioc_1.40.0        scales_1.1.1          
[52] hms_1.1.1              promises_1.2.0.1       parallel_4.1.2        
[55] yaml_2.2.1             stringi_1.7.6          highr_0.9             
[58] rlang_0.4.12           pkgconfig_2.0.3        systemfonts_1.0.3     
[61] bitops_1.0-7           evaluate_0.14          lattice_0.20-45       
[64] labeling_0.4.2         cowplot_1.1.1          processx_3.5.2        
[67] tidyselect_1.1.1       parallelly_1.30.0      magrittr_2.0.1        
[70] R6_2.5.1               generics_0.1.1         DelayedArray_0.20.0   
[73] DBI_1.1.2              mgcv_1.8-38            pillar_1.6.4          
[76] haven_2.4.3            whisker_0.4            withr_2.4.3           
[79] abind_1.4-5            RCurl_1.98-1.5         future.apply_1.8.1    
[82] modelr_0.1.8           crayon_1.4.2           car_3.0-12            
[85] utf8_1.2.2             tzdb_0.2.0             rmarkdown_2.11        
[88] locfit_1.5-9.4         grid_4.1.2             readxl_1.3.1          
[91] callr_3.7.0            git2r_0.29.0           reprex_2.0.1          
[94] digest_0.6.29          webshot_0.5.2          httpuv_1.6.5          
[97] munsell_0.5.0          viridisLite_0.4.0     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
